src/nvim/statusline.c

```diff
@@ -451,14 +427,25 @@ void win_redr_custom(win_T *wp, bool draw_winbar, bool draw_ruler)
     // Use 'tabline'.  Always at the first line of the screen.
     stl = p_tal;
     row = 0;
+    opt = "tabline";
     fillchar = ' ';
     attr = HL_ATTR(HLF_TPF);
     maxwidth = Columns;
-    use_sandbox = was_set_insecurely(wp, "tabline", 0);
+  } else if (row != -1) {
+    // Use 'numberformat' on the passed row.
+    nuf = true;
+    stl = wp->w_p_nuf;
+    col = wp->w_wincol + wp->w_scwidth + compute_foldcolumn(wp, 0);
```

Comment:
Users have also showed interest in customizing the foldcolumn. Might be possible with this as well? Not sure if an API exists to get the fold depth for a current line but I would guess it does. Anyways we shift by the sign/column width here to make sure we don't draw over it.